plugins/modules/mso_schema.py

```diff
@@ -81,41 +103,59 @@ def main():
     argument_spec = mso_argument_spec()
     argument_spec.update(
         schema=dict(type="str", aliases=["name"]),
-        # messages=dict(type='dict'),
-        # associations=dict(type='list'),
-        # health_faults=dict(type='list'),
-        # references=dict(type='dict'),
-        # policy_states=dict(type='list'),
-        state=dict(type="str", default="query", choices=["absent", "query"]),
+        id=dict(type="str"),
+        description=dict(type="str"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
     )
 
     module = AnsibleModule(
         argument_spec=argument_spec,
         supports_check_mode=True,
         required_if=[
             ["state", "absent", ["schema"]],
+            ["state", "present", ["schema"]],
         ],
     )
 
     schema = module.params.get("schema")
+    schema_id = module.params.get("id")
+    description = module.params.get("description")
     state = module.params.get("state")
 
     mso = MSOModule(module)
-
-    schema_id = None
     path = "schemas"
 
     # Query for existing object(s)
     if schema:
-        mso.existing = mso.get_obj(path, displayName=schema)
+        if schema_id:
+            mso.existing = mso.get_obj(path, id=schema_id)
+        else:
+            mso.existing = mso.get_obj(path, displayName=schema)
+
         if mso.existing:
-            schema_id = mso.existing.get("id")
+            if not schema_id:
+                schema_id = mso.existing.get("id")
             path = "schemas/{id}".format(id=schema_id)
     else:
         mso.existing = mso.query_objs(path)
 
-    if state == "query":
-        pass
+    mso.previous = mso.existing
+    if state == "present":
+        mso.sanitize(dict(displayName=schema, id=schema_id, description=description), collate=True)
+        if mso.existing:
+            ops = []
+            if mso.existing.get("displayName") != schema:
+                ops.append(dict(op="replace", path="/displayName", value=schema))
+            if mso.existing.get("description") != description and description is not None:
+                ops.append(dict(op="replace", path="/description", value=description))
+
+            if not module.check_mode:
+                mso.request(path, method="PATCH", data=ops)
+        else:
+            if not module.check_mode:
+                # Initialize templates as empty list, otherwise mso_schema_template will fail to add templates
+                mso.request(path, method="POST", data=dict(displayName=schema, description=description, templates=[]))
```

Comment:
Do older versions of MSO accept the templates=[]?

Reply:
Have removed, because this was not needed to be initialised on NDO 4.2+. Older version should not be impacted anyway because it is not status=present if not supported by our module ( defined in docs ).

For NDO4.1 there would still be an issue with empty schema's when adding templates via PATCH. This is because in that version NDO doesn't support the PATCH operation yet and you would see following error: "message": "add operation does not apply: doc is missing path: \"/templates/-\": missing value". Basically NDO does not initialise the empty array and thus not allow for add to the array.